wrappers/Julia/CoolProp.jl

```diff
@@ -1,190 +1,677 @@
-VERSION < v"0.4.0" && __precompile__()
```

Comment:
To find what is the best practice for  `__precompile__()` I had a survey by checking many of the most favorite Julia packages. (also using [github search result](https://github.com/search?utf8=%E2%9C%93&q=__precompile__&type=Code&ref=searchresults) is helpful) , results show that most of package developers do not care about old Julia versions e.g.: 
1. [HDF5](https://github.com/JuliaIO/HDF5.jl/blob/efc4a3ae5cca7e46d246fe0e1e305e4dd3a8b0e8/src/HDF5.jl#L1)
2. [ACME](https://github.com/HSU-ANT/ACME.jl/blob/develop/src/ACME.jl#L4)
3. [ApproxFun](https://github.com/JuliaApproximation/ApproxFun.jl/blob/development/src/ApproxFun.jl#L1)
4. MixedModels.jl
5. NLopt.jl
6. Optim.jl
7. Plots.jl
8. PyPlot.jl
9. QuantEcon.jl
10. RCall.jl
11. Reactive.jl
12. Requests.jl
13. StatsBase.jl
14. Sundials.jl
15. TensorFlow.jl
16. Weave.jl

or even don't add `__precompile__()` on master (some had commented it) (and I didn't look at tags):

1. ArrayFire.jl
2. MachineLearning.jl
3. Mocha.jl
4. Morsel.jl
5. Mux.jl
6. AudioIO.jl
7. OnlineStats.jl
8. OpenCL.jl
9. Orchestra.jl
10. ParallelAccelerator.jl
11. Playground.jl
12. RDatasets.jl
13. Reel.jl
14. TextAnalysis.jl
15. TypeCheck.jl
16. XGBoost.jl

IMO, `__precompile__()` make it fast for users but [buggy for developers](http://stackoverflow.com/questions/40116045/why-is-julia-taking-a-long-time-on-the-first-call-into-my-module/40140349#40140349).  
So I prefer to add a comment on master and let it happen on release. In this case we will need to look for a tagged suitable CoolProp wrapper when he `Pkg.add("CoolProp")`. (maybe by using same tag numbers in both Repos).